### PR TITLE
NIRISS SOSS assign_wcs

### PR DESCRIPTION
**Pull request by brechmos-stsci**
In discussion with @nden , this is the current code for NIRISS SOSS assign_wcs.  I can assign a WCS to the input data based on the wavelength lookup file. We are not sure if the input science file and wavelength lookup file are consistent with each other and we will look into that when we have more information from the NIRISS team.

@nden @hbushouse

### CODE REVIEW

File: jwst/assign_wcs/niriss.py [hunk 1]
```diff
@@ -1,26 +1,116 @@
...
+    """
+    The NIRISS SOSS pipeline includes 3 coordinate frames -
+    detector, focal plane and sky
+
+    reference_files={'distortion': 'soss_wavelengths_configuration.asdf'}
+    """
+
+    # Get the target RA and DEC, they will be used for setting the WCS RA and DEC based on a conversation
+    # with Kevin Volk.
+    target_ra, target_dec = 0.0, 0.0
```
Comment (nden) [inline jwst/assign_wcs/niriss.py]: Is this assignment necessary, given that an error is raised if they are not available in the model.

File: jwst/assign_wcs/niriss.py [hunk 2]
```diff
@@ -1,26 +1,116 @@
...
+    try:
+        target_ra = float(input_model['meta.target.ra'])
+        target_dec = float(input_model['meta.target.dec'])
+    except:
+        # There was an error getting the target RA and DEC, so we are not going to continue.
+        raise ValueError('Problem getting the TARG_RA or TARG_DEC from input model {}'.format(input_model))
+
+    # Define the frames
+    detector = cf.Frame2D(name='detector', axes_order=(0, 1), unit=(u.pix, u.pix))
+    sky = cf.CelestialFrame(reference_frame=coord.ICRS())
```
Comment (nden) [inline jwst/assign_wcs/niriss.py]: Add `units` (deg), `axes_names` and `axes_order`.

File: jwst/assign_wcs/niriss.py [hunk 3]
```diff
@@ -1,26 +1,116 @@
...
+    Parameters
+    ----------
+    reference_files - Currently should be just a single file.
+
+    Returns
+    -------
+    wavelength_lookup - Return the Tabular2D lookup model.
+
+    """
+    wavelength_lookup = AsdfFile.open(reference_files['distortion'])
```
Comment (nden) [inline jwst/assign_wcs/niriss.py]: The wavelength correction will be of type `specwcs`.

File: jwst/assign_wcs/niriss.py [hunk 4]
```diff
@@ -1,26 +1,116 @@
...
+    return wavelength_lookup
```
Comment (nden) [inline jwst/assign_wcs/niriss.py]: Hm, is having this in a separate function going to leave an open file?

File: jwst/assign_wcs/niriss.py [hunk 5]
```diff
@@ -1,26 +1,116 @@
...
+
+
+def niriss_soss(input_model, reference_files):
+    """
+    The NIRISS SOSS pipeline includes 3 coordinate frames -
+    detector, focal plane and sky
+
+    reference_files={'distortion': 'soss_wavelengths_configuration.asdf'}
```
Comment (nden) [inline jwst/assign_wcs/niriss.py]: Change the reference type to `specwcs`. `distortion` is used for spatial distortion reference data.

File: jwst/assign_wcs/niriss.py (hunk 2 above)
Comment (brechmos-stsci) [inline jwst/assign_wcs/niriss.py]: I can add that but I looked at some of the other scripts (e.g., miri.py) and none of them have `axes_order`, `axes_names`, nor `units` for they sky

Comment (nden) [inline jwst/assign_wcs/niriss.py]: Well, they should.

**Comment by brechmos-stsci** [comment]
Trying to track down a "Segmentation Fault: 11" introduced after updating my environment.

**Comment by brechmos-stsci** [comment]
I found and fixed the segmentation fault.  Theoretically I squashed the commits.

**Pull request merged**